notebooks/02-Summary-statistics-and-plots.ipynb

```diff
@@ -0,0 +1,512 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cf4d92f0-b624-452c-9f51-52fe70dfde88",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df[df[\"circular\"] == False][\"end_to_end_distance\"].plot.hist(\n",
+    "    figsize=(16, 9), bins=20, title=\"End to End Distance\", alpha=0.5\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94270659-cc2a-4606-9fa9-80ed5fac82d2",
+   "metadata": {},
+   "source": [
+    "### Multiple Images\n",
+    "\n",
+    "Often you will have processed multiple images and you will want to plot the distributions of metrics for each image\n",
+    "separately.\n",
+    "\n",
+    "For this example we duplicate the data and append it, adjusting the values slightly"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "529aedb3-2d4e-4703-86a0-bcba99caee57",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def scale_df(df: pd.DataFrame, scale: float, image: str) -> pd.DataFrame:\n",
+    "    \"\"\"Scale the numerical values of a data frame. Retains string variables and the index.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    df: pd.DataFrame\n",
+    "        Pandas Dataframe to scale.\n",
+    "    scale: float\n",
+    "        Factor by which to scale the data.\n",
+    "    image: str\n",
+    "        Name for new (dummy) image.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    pd.DataFrame\n",
+    "        Scaled data frame\n",
+    "    \"\"\"\n",
+    "    _df = df[df.select_dtypes(include=[\"number\"]).columns] * scale\n",
+    "    _df.reset_index(inplace=True)\n",
+    "    _df[\"image\"] = image\n",
+    "    _df = pd.concat([_df, df[[\"circular\", \"basename\"]]], axis=1)\n",
+    "    _df.set_index([\"image\", \"threshold\", \"molecule_number\"], inplace=True)\n",
+    "    return _df\n",
+    "\n",
+    "\n",
+    "smaller = scale_df(df, scale=0.9, image=\"smaller\")\n",
+    "larger = scale_df(df, scale=1.25, image=\"larger\")\n",
+    "df_three_images = pd.concat([smaller, df, larger])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6a57eb1a-3093-4ca6-8411-e62ea2614730",
+   "metadata": {},
+   "source": [
+    "### Contour Length from Three Processed Images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9820d259-680e-49fd-a1eb-e14d200a0b7a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_three_images[\"contour_lengths\"].groupby(\"image\").plot.hist(\n",
+    "    figsize=(16, 9),\n",
+    "    bins=20,\n",
+    "    # bins=np.linspace(start, stop, num=x), # Use this if there are different number of grains per image\n",
+    "    title=\"Contour Lengths\",\n",
+    "    alpha=0.5,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c502a69c-79ec-467c-964a-81b3f3b87913",
+   "metadata": {},
+   "source": [
+    "### Ignoring Image\n",
```

Comment:
```suggestion
    "### Ignoring Images\n",
```

Reply:
Its not a case of ignoring images here, rather its ignoring the `.groupby("image")` option which meant things were plotted per-image. The plural suggests, to me, that a subset are being ignored.